core/node/node_sync/src/client.rs

```diff
@@ -105,7 +105,25 @@ impl MainNodeClient for Box<DynClient<L2>> {
     }
 
     async fn fetch_genesis_config(&self) -> EnrichedClientResult<GenesisConfig> {
-        self.genesis_config().rpc_context("genesis_config").await
+        let dto = self.genesis_config().rpc_context("genesis_config").await?;
+        Ok(GenesisConfig {
```

Comment:
Why do we need both DTO and non dto?

Reply:
Single responsibility principle; it's *weird* to have a single type represent both a configuration and something used by API. More practically, it makes `zksync_web3_decl` depend on `zksync_config`, which is unnecessary and makes zkstack transitively depend on `zksync_config` as well (since it depends on `zksync_web3_decl`), even if the direct dependency is removed.